src/libs/actions/IOU.ts

```diff
@@ -926,7 +926,8 @@ function getMoneyRequestInformation(
         false,
         currentTime,
     );
-    const optimisticTransactionThread = ReportUtils.buildTransactionThread(iouAction, iouReport.reportID);
+    const optimisticTransactionThread = ReportUtils.buildTransactionThread(iouAction, iouReport);
+    iouAction.childReportID = optimisticTransactionThread.reportID;
```

Comment:
Should we add a comment above this line saying why we need to do this? It's not super obvious right away, and it made me wonder if we should be setting this in `const iouAction = ReportUtils.buildOptimisticIOUReportAction`, but then realized `ReportUtils.buildTransactionThread` takes `iouAction` which is created by `ReportUtils.buildOptimisticIOUReportAction`.

I don't have a good solution for that right now -- it seems like we could generate an optimistic reportID before the calls to `ReportUtils.buildOptimisticIOUReportAction` and `ReportUtils.buildTransactionThread`, and pass it to both so we don't have to manually set `iouAction.childReportID = optimisticTransactionThread.reportID;`, but I'm not sure if that's a good idea.

I'm curious what you think

Reply:
Yes, unfortunately these 2 are co-dependent: the transaction thread has the iou action's ID as the parent action ID, and vice versa. Also, passing the pre-generated report ID will cause a lot of refactoring of the current methods that use the `buildOptimisticChatReport`. One possible idea here is to make a separate function that will generate the `iouAction`, `optimisticTransactionThread`, and the `optimisticCreatedActionForTransactionThread`. This way, we'll not be setting the `iouAction.childReportID` in 4 different places, but in only one function, and will be calling it from those 4 places. It's not perfect either, but will minimize code duplication.

Smth like:

```js
function buildOptimisticIOUReportActionWithTransactionThread(...) {
  const iouAction = ReportUtils.buildOptimisticIOUReportAction(
        CONST.IOU.REPORT_ACTION_TYPE.CREATE,
        amount,
        currency,
        comment,
        [participant],
        optimisticTransaction.transactionID,
        undefined,
        iouReport.reportID,
        false,
        false,
        receiptObject,
        false,
        currentTime,
    );
    const optimisticTransactionThread = ReportUtils.buildTransactionThread(iouAction, iouReport.reportID);
    iouAction.childReportID = optimisticTransactionThread.reportID;
    const optimisticCreatedActionForTransactionThread = ReportUtils.buildOptimisticCreatedReportAction(payeeEmail);


  return { iouAction, optimisticTransactionThread, optimisticCreatedActionForTransactionThread }
}
```

Reply:
Yeah I thought of that but wasn't convinced it would be better. But on second thought, I do think it's better and we can add a comment inside that function above where we set `iouAction.childReportID` so people know not to remove it.

The alternative would be to put that comment in 4 places, and we'd have that code duplicated like you said, and nothing prevents people from forgetting to set `iouAction.childReportID`

I'm good to update to this 👍

Reply:
Done ✔️ 

@situchan I've re-tested the 4 different flows that use the 4 different functions that call this new `buildOptimisticMoneyRequestEntities` function – everything works as expected (except for the known Complete Split BE issue):
- Send Money
- Request Money
- Split right away (the fixed amount split)
- Split with Complete Split (the scan split)